src/docfx/lib/moniker/MonikerRangeParser.cs

```diff
@@ -3,42 +3,64 @@
 
 using System;
 using System.Collections.Concurrent;
+using System.Collections.Generic;
 using System.Linq;
 
 namespace Microsoft.Docs.Build
 {
     internal class MonikerRangeParser
     {
-        private readonly ConcurrentDictionary<string, string[]> _cache = new ConcurrentDictionary<string, string[]>(StringComparer.OrdinalIgnoreCase);
+        private readonly ConcurrentDictionary<string, (List<Error>, string[])> _cache = new ConcurrentDictionary<string, (List<Error>, string[])>(StringComparer.OrdinalIgnoreCase);
         private readonly EvaluatorWithMonikersVisitor _monikersEvaluator;
 
         public MonikerRangeParser(EvaluatorWithMonikersVisitor monikersEvaluator)
```

Comment:
```suggestion
        public MonikerRangeParser(MonikerDefinitionModel monikerDefinition)
```

Reply:
`EvaluatorWithMonikersVisitor ` is used in line 58 as well.
Refactoring the implementation of parser is not the purpose of this PR

Reply:
I mean constructing `EvaluatorWithMonikersVisitor` inside the constructor of `MonikerRangeParser`. `MonikerProvider.cs` line 40 can become `new MonikerRangeParser(monikerDefinition)` and can remove `MonikerProvider.Comparer`.

Reply:
Get your point now, the construction of `EvaluatorWithMonikersVisitor` is duplicated now in all the callers

Reply:
#5737